### PR TITLE
refactor(artifacts): replace list of depNames with PackageDependency list

### PR DESCRIPTION
**Pull request by secustor**
## Changes:

This PR replaces the `depName` array function parameter of the `updateArtifacts` functions with an array of type `PackageDependency`.  

## Context:

The changes are introduced as preparation for the implementation of the support for Terraform community providers.
 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

File: lib/manager/composer/__snapshots__/artifacts.spec.ts.snap [hunk 1]
```diff
@@ -73,7 +73,7 @@ Array [
 exports[`.updateArtifacts() returns null if unchanged 1`] = `
 Array [
   Object {
-    "cmd": "composer update --with-dependencies --ignore-platform-reqs --no-ansi --no-interaction",
+    "cmd": "composer update foo bar --with-dependencies --ignore-platform-reqs --no-ansi --no-interaction",
```
Comment (viceice) [inline lib/manager/composer/__snapshots__/artifacts.spec.ts.snap:76]: 🤔 strange change

**Review by viceice — COMMENTED** [review]
Can you explain a little bit more why this is needed?

File: lib/manager/composer/__snapshots__/artifacts.spec.ts.snap (hunk 1 above)
Comment (secustor) [reply lib/manager/composer/__snapshots__/artifacts.spec.ts.snap:76]: I had to add depNames to a composer test  because of the codecoverage

**Comment by secustor** [comment]
For tje extension of the lock file implementation,  I need access to the lookupName and registryUrls as depNames can collide there. 

E.g. a community provider for aws and a community provided one.

**Review by viceice — APPROVED** [review]

**Pull request merged**